### PR TITLE
plasma-5,kde,kde-frameworks: update

### PR DESCRIPTION
**Pull request by bachp**
###### Motivation for this change

This is still work in progress in case somebody else is interested in helping out.

Update Plasma, KDE frameworks and KDE Applications to the newest versions:

- Plasma: 5.18.3
- KDE Framworks: 5.68
- KDE Applications: 19.12.3

This is the first time I'm working with KDE packaging in NixOS so I'm not sure I did it right.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/plasma-5/kde-gtk-config/default.nix [hunk 1]
```diff
@@ -13,7 +13,6 @@ mkDerivation {
     ki18n kio glib gtk2 gtk3 karchive kcmutils kconfigwidgets kiconthemes
     knewstuff gsettings-desktop-schemas
   ];
-  patches = [ ./patches/follow-symlinks.patch ./patches/gsettings.patch ];
```
Comment (ttuegel) [inline pkgs/desktops/plasma-5/kde-gtk-config/default.nix]: These patches are essential; where did they go?

File: pkgs/desktops/plasma-5/plasma-workspace/0001-startkde.patch [hunk 2]
```diff
@@ -48,21 +48,6 @@ index e0f7004..8ac41fd 100644
  }
  
  QStringList allServices(const QLatin1String& prefix)
-@@ -184,14 +184,6 @@ void runEnvironmentScripts()
-         }
-     }
-     sourceFiles(scripts);
--
```
Comment (ttuegel) [inline pkgs/desktops/plasma-5/plasma-workspace/0001-startkde.patch]: It is important that `startkde` doesn't contaminate the `XDG_DATA_DIRS` variable. Why is this removed?

File: pkgs/desktops/plasma-5/plasma-workspace/default.nix [hunk 3]
```diff
@@ -44,7 +44,7 @@ mkDerivation {
 
   patches = [
     ./0001-startkde.patch
-    ./0002-absolute-wallpaper-install-dir.patch
+#    ./0002-absolute-wallpaper-install-dir.patch #TODO(bachp): check if this is needed https://github.com/KDE/plasma-workspace/blob/Plasma/5.18/sddm-theme/theme.conf.cmake
```
Comment (ttuegel) [inline pkgs/desktops/plasma-5/plasma-workspace/default.nix]: This patch is needed to set the correct absolute path to the SDDM theme wallpaper.

File: pkgs/development/libraries/kde-frameworks/kcmutils/series [hunk 4]
```diff
@@ -1,2 +1 @@
 kcmutils-follow-symlinks.patch
-kcmutils-debug-module-loader.patch
```
Comment (ttuegel) [inline pkgs/development/libraries/kde-frameworks/kcmutils/series]: This patch is important for us to get useful bug reports. Please leave it in place.

**Review by ttuegel — CHANGES_REQUESTED** [review]
I don't know what's going on with the patches, but some important fixes were removed.

Because these are three package sets with different release schedules, I would prefer to update them each in a separate pull request.

File: pkgs/development/libraries/kde-frameworks/kcmutils/series (hunk 4 above)
Comment (bachp) [reply pkgs/development/libraries/kde-frameworks/kcmutils/series]: If I recall correctly the files this was patching does no longer exists.

Comment (ttuegel) [reply pkgs/development/libraries/kde-frameworks/kcmutils/series]: We need to find out where that code went and apply the patch there. It may even have moved to a different library, unfortunately.

Comment (bachp) [reply pkgs/development/libraries/kde-frameworks/kcmutils/series]: I just tried to get things compiling so we know what to look at. But the patches are not very well documented so it's hared to tell why they are needed.

How is this normaly done?

**Comment by peterhoeg** [comment]
Superseded by #84542 and #83897

**Comment by peterhoeg** [comment]
Just closing this may seem a bit ungrateful - sorry if it seemed that way @bachp.

Thanks for doing this but I suggest you help out on the linked PR instead.

**Comment by petabyteboy** [comment]
Sorry, I didn't see this one before. My result is mostly the same until now, but I am still working on re-adding all the patches.

**Comment by bachp** [comment]
@peterhoeg No worries, I'm actually glad somebody more experienced takes over, I can look over their shoulders.
